app/components/osc-webhook-triggers/osc-webhook-triggers.html

```diff
@@ -1,38 +1,53 @@
-<ng-form name="secretsForm" class="add-webhook">
+<ng-form name="$ctrl.secretsForm" class="add-webhook">
   <div ng-repeat="trigger in $ctrl.webhookTriggers">
     <div class="add-webhook-row">
-      <ui-select 
-        ng-model="trigger.data.type"
-        ng-disabled="$ctrl.isDeprecated(trigger)"
-        on-select="$ctrl.triggerTypeChange(trigger)"
-        search-enabled="false"
-        title="Select a webhook type"
-        class="select-webhook-type"
-        flex>
-        <ui-select-match placeholder="Webhook type">
-          {{ $select.selected.label }}
-        </ui-select-match>
-        <ui-select-choices repeat="option.label as option in $ctrl.webhookTypesOptions">
-          {{ option.label }}
-        </ui-select-choices>
-      </ui-select>
+      <div class="select-webhook-type">
+        <ui-select 
+          ng-model="trigger.data.type"
+          name="triggerType{{$index}}"
+          ng-disabled="$ctrl.isDeprecated(trigger)"
+          on-select="$ctrl.triggerTypeChange(trigger)"
+          search-enabled="false"
+          title="Select a webhook type"
+          ng-class="{'has-error': $ctrl.secretsForm.triggerType{{$index}}.$touched && !trigger.data.type && trigger.data[trigger.data.type.toLowerCase()].secretReference.name }"
+          focus-on="triggerTypeFocus{{$index}}"
```

Comment:
I don't think `triggerTypeFocus` exists as a function, though I see you broadcasting an `event` [up here](https://github.com/openshift/origin-web-console/pull/2689/files#diff-9d51eb270d91c5c6831894971b883c5cR107).  I can't quite tell that it is working.  It looks like the example just expects a [callback function](https://github.com/angular-ui/ui-select/wiki/ui-select#examples-set-focus).

Reply:
It should work, when user clicks on the `Add webhook` the new trigger type selectbox should be focused. Not really sure whats the issue, `focus-on` should take an event as a parameter. In the example ng-click triggers a function which calls the broadcast on the element with desired `focus-on` string.

Reply:
Ok cool, wanted to ask & make sure.  Have we used events like this before?  I wonder if it would be good to have a naming convention for the variable.